File: 0002-add-two-numbers/0002-add-two-numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        dummy=ListNode()
        cur=dummy
        mark=0
        while l1 or l2 or mark:
            x = l1.val if l1 else 0
            y = l2.val if l2 else 0

            temp = x + y + mark
            cur.next = ListNode(temp % 10)
            mark = temp // 10

            cur = cur.next

            if l1:
                l1 = l1.next
            if l2:
                l2 = l2.next

        return dummy.next

File: 0002-add-two-numbers/0002-add-two-numbers.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func addTwoNumbers(l1 *ListNode, l2 *ListNode) *ListNode {
    dummy:=&ListNode{}
    cur:=dummy
    carry:=0
    for l1!=nil || l2!=nil || carry !=0{
        x,y:=0,0
        if l1!=nil{
            x=l1.Val
            l1=l1.Next
        }
        if l2!=nil{
            y=l2.Val
            l2=l2.Next
        }
        sum:=x+y+carry
        carry=sum/10
        cur.Next=&ListNode{Val:sum%10}
        cur=cur.Next
    }
    return dummy.Next
}